src/ResourceManagement/Network/AzureFirewallImpl.cs

```diff
@@ -0,0 +1,1298 @@
+        }
+
+        private AzureFirewallImpl DefineAzureFirewallNetworkRule(string name)
+        {
+            networkRule = new AzureFirewallNetworkRule
+            {
+                Name = name
+            };
+            if (networkRuleCollection.Rules == null)
+            {
+                networkRuleCollection.Rules = new List<AzureFirewallNetworkRule>();
+            }
+            return this;
+        }
+
+        private AzureFirewallImpl DefineAzureFirewallNetworkRuleCollection(string name)
+        {
+            networkRuleCollection = new AzureFirewallNetworkRuleCollectionInner
+            {
+                Name = name
+            };
+            if (Inner.NetworkRuleCollections == null)
+            {
+                Inner.NetworkRuleCollections = new List<AzureFirewallNetworkRuleCollectionInner>();
+            }
+            return this;
+        }
+
+        private AzureFirewallImpl WithAdditionalProperties(IDictionary<string, string> additionalProperties)
+        {
+            Inner.AdditionalProperties = additionalProperties;
+            return this;
+        }
+
+        private AzureFirewallImpl WithAdditionalProperty(string key, string value)
+        {
+            Inner.AdditionalProperties.Add(key, value);
+            return this;
+        }
+
+        private AzureFirewallImpl WithoutAdditionalProperty(string key)
+        {
+            if (Inner.AdditionalProperties != null)
+            {
+                Inner.AdditionalProperties.Remove(key);
+            }
+            return this;
+        }
+
+        private AzureFirewallImpl WithAlertModeForThreatIntel()
+        {
+            Inner.ThreatIntelMode = AzureFirewallThreatIntelMode.Alert;
+            return this;
+        }
+
+        private AzureFirewallImpl WithAllowActionTypeInApplicationRuleCollection()
+        {
+            applicationRuleCollection.Action = new AzureFirewallRCAction(AzureFirewallRCActionType.Allow);
+            return this;
+        }
+
+        private AzureFirewallImpl WithAllowActionTypeInNetworkRuleCollection()
+        {
+            networkRuleCollection.Action = new AzureFirewallRCAction(AzureFirewallRCActionType.Allow);
+            return this;
+        }
+
+        private AzureFirewallImpl WithAnyProtocol()
+        {
+            natRule.Protocols.Add(AzureFirewallNetworkRuleProtocol.Any);
+            return this;
+        }
+
+        private AzureFirewallImpl WithAzfwHubSkuName()
+        {
+            if (Inner.Sku == null)
+            {
+                Inner.Sku = new AzureFirewallSku();
+            }
+            Inner.Sku.Name = AzureFirewallSkuName.AZFWHub;
+            return this;
+        }
+
+        private AzureFirewallImpl WithAzfwVnetSkuName()
+        {
+            if (Inner.Sku == null)
+            {
+                Inner.Sku = new AzureFirewallSku();
+            }
+            Inner.Sku.Name = AzureFirewallSkuName.AZFWVNet;
+            return this;
+        }
```

Comment:
Should we name it e.g WithAzureFirewallHubSku (Azfw -> full name, drop the Name since we know it is SKU)?
I am not sure about the background on VNet or Hub. So this suggestion might be wrong.
@yaohaizh